### ISSUE TITLE
[Project Addition]:Weather Prediction

### ISSUE THREAD

**Issue by manishh12** [abhisheks008/DL-Simplified#602]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Weather Prediction</br>
:red_circle: **Aim** : We are going to predict the weather condition is good for bbq based on different features</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/thedevastator/weather-prediction/data?select=weather_prediction_dataset.csv</br>https://www.kaggle.com/datasets/thedevastator/weather-prediction/data?select=weather_prediction_bbq_labels.csv</br>
:red_circle: **Approach** :  I will be implementing Artificial Neural Network,Convolutional Neural Network and Decision Tree or Random Forest Algo.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Manish Kumar Gupta
- GitHub Profile Link : https://github.com/manishh12
- Email ID :manishdid360@gmail.com
- Participant ID (if applicable):manishh12
- Approach for this Project :
Data Loading and Exploration: I will load the dataset containing weather measurements from the dataset I will explore the dataset to understand the features and labels provided.
Data Visualization: I will visualize the relationship between different weather features and the label using kernel density estimation (KDE) plots and pie charts to gain insights into the distribution and proportion of suitable and unsuitable weather conditions.
Data Preparation: I will prepare the dataset for model training by separating features and labels. Then, I will split the data into training and testing sets.
Model Building:
Artificial Neural Network (ANN): I will build a simple ANN with two dense layers and train it using the training data. Then, I will evaluate the model's performance using accuracy metrics and a confusion matrix.
Convolutional Neural Network (CNN): As an alternative approach, I will construct a CNN with a single Conv1D layer, reshaping the input data appropriately. Similar to ANN, I will evaluate its performance.
Random Forest Classifier: I will employ a Random Forest model and train it on the data. Then, I will evaluate the accuracy of the model.
Results Visualization: I will compare the accuracies of the three models (ANN, CNN, and Random Forest) using a bar plot to determine which method performs better for predicting barbecue weather conditions in Dresden.
- What is your participant role? (Mention the Open Source program)Contributor in GSSOC'24

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by dasdebanna** [comment]
Hi @manishh12 I want to work on this issue. Can you assign me this issue and I'm participating in GSSOC'24?

**Comment by manishh12** [comment]
> Hi @manishh12 I want to work on this issue. Can you assign me this issue and I'm participating in GSSOC'24?

I am requesting the author of this repository to assign this task to me. As a contributor in GSSOC'24, I am not proposing this as an issue for others.

**Comment by abhisheks008** [comment]
Hi @manishh12 thanks for opening this issue. Try to focus on deep learning method implementations rather than going for machine learning methods.

**Comment by abhisheks008** [comment]
You need to implement at least 2-3 methods/architectures for this issue. That's why I am asking about a brief approach.

**Comment by manishh12** [comment]
> You need to implement at least 2-3 methods/architectures for this issue. That's why I am asking about a brief approach.
Approach: 
For ANN model 
I will Create a Sequential model with three dense layers. The first layer matches the input features, the second has half the neurons of the first, and the final layer has one neuron ,with each layer with appropriate activation functions then will Compile the model using binary cross-entropy loss and the Adam optimizer and will track accuracy as the evaluation metric. After that will train the model on training and validation data for a specific number of epochs then will plot training and validation loss and accuracy over epochs and plot a confusion matrix to assess the model's predictive performance.

For CNN :
I will first transform the input data into the required format as I will be using Conv1D layers, then will Utilize a Sequential model, starting with a Reshape layer to reshape the input data. Following this with a Conv1D layer featuring with appropriate features and kernel. Then the output from this convolutional layer will be flattened. Addition of Dense layer with one neuron and a sigmoid activation function for binary classification. Then will Compile the model using binary cross-entropy loss and the Adam optimizer, while employing accuracy as a metric for evaluation and then train the model for multiple epochs using both training and validation datasets after this will make predictions on the test data and generate a classification report to evaluate precision, recall, and F1-score and will plot a confusion matrix to visually assess the model's performance in prediction.

**Comment by abhisheks008** [comment]
Looks good to me! Assigning this issue to you @manishh12
